bundles/org.openhab.binding.jablotron/src/main/java/org/openhab/binding/jablotron/internal/discovery/JablotronDiscoveryService.java

```diff
@@ -0,0 +1,172 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.*;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.openhab.binding.jablotron.internal.handler.JablotronBridgeHandler;
+import org.openhab.binding.jablotron.internal.model.JablotronDiscoveredService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JablotronDiscoveryService} is responsible for the thing discovery
+ * process.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+@NonNullByDefault
+public class JablotronDiscoveryService extends AbstractDiscoveryService
+        implements DiscoveryService, ThingHandlerService {
+    private final Logger logger = LoggerFactory.getLogger(JablotronDiscoveryService.class);
+
+    private @Nullable JablotronBridgeHandler bridgeHandler;
+
+    private @Nullable ScheduledFuture<?> discoveryJob = null;
+
+    public JablotronDiscoveryService() {
+        super(DISCOVERY_TIMEOUT_SEC);
+        logger.debug("Creating discovery service");
+    }
+
+    private void startDiscovery() {
+        if (bridgeHandler != null && ThingStatus.ONLINE == bridgeHandler.getThing().getStatus()) {
+            discoverServices();
+        }
+    }
+
+    @Override
+    public void setThingHandler(@Nullable ThingHandler thingHandler) {
+        if (thingHandler instanceof JablotronBridgeHandler) {
+            bridgeHandler = (JablotronBridgeHandler) thingHandler;
+        }
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return bridgeHandler;
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        super.stopBackgroundDiscovery();
+        if (discoveryJob != null) {
+            discoveryJob.cancel(true);
+        }
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Starting Jablotron background discovery");
+
+        if (discoveryJob == null || discoveryJob.isCancelled()) {
+            discoveryJob = scheduler.scheduleWithFixedDelay(this::startDiscovery, 10, 3600, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void activate() {
+        super.activate(null);
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting scanning for items...");
+        startDiscovery();
+    }
+
+    public void oasisDiscovered(String label, String serviceId) {
+        ThingUID thingUID = new ThingUID(THING_TYPE_OASIS, bridgeHandler.getThing().getUID(), serviceId);
```

Comment:
Here too and same for below.
```suggestion
        JablotronBridgeHandler localBridgeHandler = bridgeHandler;
        if (localBridgeHandler != null) {
            ThingUID thingUID = new ThingUID(THING_TYPE_OASIS, localBridgeHandler.getThing().getUID(), serviceId);
```

Reply:
ok, done